app/main.cpp

```diff
@@ -312,33 +312,34 @@ void initDeclarative()
   qmlRegisterType< BluetoothDiscoveryModel >( "lc", 1, 0, "BluetoothDiscoveryModel" );
   qmlRegisterType< PositionProvidersModel >( "lc", 1, 0, "PositionProvidersModel" );
 
-  qmlRegisterUncreatableType< QgsUnitTypes >( "QgsQuick", 0, 1, "QgsUnitTypes", "Only enums from QgsUnitTypes can be used" );
-  qmlRegisterType< QgsVectorLayer >( "QgsQuick", 0, 1, "VectorLayer" );
-  qmlRegisterType< QgsProject >( "QgsQuick", 0, 1, "Project" );
-  qmlRegisterType< QgsQuickMapCanvasMap >( "QgsQuick", 0, 1, "MapCanvasMap" );
-  qmlRegisterType< QgsQuickMapSettings >( "QgsQuick", 0, 1, "MapSettings" );
-  qmlRegisterType< QgsQuickMapTransform >( "QgsQuick", 0, 1, "MapTransform" );
-  qmlRegisterType< QgsQuickCoordinateTransformer >( "QgsQuick", 0, 1, "CoordinateTransformer" );
+  qmlRegisterUncreatableType< QgsUnitTypes >( "Input", 0, 1, "QgsUnitTypes", "Only enums from QgsUnitTypes can be used" );
+  qmlRegisterType< QgsVectorLayer >( "Input", 0, 1, "VectorLayer" );
+  qmlRegisterType< QgsProject >( "Input", 0, 1, "Project" );
+  qmlRegisterType< InputMapCanvasMap >( "Input", 0, 1, "MapCanvasMap" );
+  qmlRegisterType< InputMapSettings >( "Input", 0, 1, "MapSettings" );
+  qmlRegisterType< InputMapTransform >( "Input", 0, 1, "MapTransform" );
+  qmlRegisterType< InputCoordinateTransformer >( "Input", 0, 1, "CoordinateTransformer" );
 
   qmlRegisterUncreatableType< AbstractPositionProvider >( "lc", 1, 0, "PositionProvider", "Must be instantiated via its construct method" );
 
   // map tools
   qmlRegisterUncreatableType< AbstractMapTool >( "lc", 1, 0, "AbstractMapTool", "Instantiate one of child map tools instead" );
   qmlRegisterType< RecordingMapTool >( "lc", 1, 0, "RecordingMapTool" );
   qmlRegisterType< SplittingMapTool >( "lc", 1, 0, "SplittingMapTool" );
-
-  qmlRegisterType( QUrl( "qrc:/qgsquickmapcanvas.qml" ), "QgsQuick", 0, 1, "MapCanvas" );
 }
 
 void addQmlImportPath( QQmlEngine &engine )
 {
-  // This adds a runtime qml directory containing QgsQuick plugin
+  // QML resources
+  engine.addImportPath( ":/com.merginmaps/imports:" );
+
+  // This adds a runtime qml directory containing Input plugin
   // when Input is installed (e.g. Android/Win32)
   engine.addImportPath( QgsApplication::qmlImportPath() );
   qDebug() << "adding QML import Path: " << QgsApplication::qmlImportPath();
 
 #ifdef QML_BUILD_IMPORT_DIR
-  // Adds a runtime qml directory containing QgsQuick plugin
+  // Adds a runtime qml directory containing Input plugin
```

Comment:
Can we remove this `#ifdef QML_BUILD_IMPORT_DIR` and `#ifdef Q_OS_IOS` (below on line 350) altogether now?

Reply:
ok

Reply:
maybe you can fix it when merged to master?